2.data_preparation.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # 2. Data preparation
+
+# Create a python script that will convert the previously generated JSON files into flattened CSV files, for each of the three exercises above. Sample files will be available in the folder sample/data
+# 
+# ### Output/Deliverables:
+# - One flattened CSV file for each of the entities (teams, players, games)
+
+# In[32]:
+
+
+# pip install pandas
+
+
+# In[33]:
+
+
+# pip install numpy
+
+
+# In[34]:
+
+
+# Data processing
+# ==============================================================================
+import pandas as pd
+import numpy as np
+import json
+
+# Display all columns
+# ==============================================================================
+pd.options.display.max_columns = None
+
+
+# In[2]:
+
+
+# This function the entire file, add brackets [] to encapsulate the string and then load it as string with json.loads()
+def read_json_file(file):
+    
+    with open(file, "r") as r:
+        response = r.read()
+        response = response.replace('\n', '')
+        response = response.replace('}{', '},{')
+        response = "[" + response + "]"
+    return json.loads(response)
+
+
+# In[30]:
+
+
+#The above code is creating a list of dictionaries and then converting it to a dataframe anda saving as a csv file.
+
+def csv_file(open_file,csv_file_name):
+   
+    information=[]
+    for call in open_file:
+        for i in call["data"]:
+            information.append(i)
+            df = pd.DataFrame.from_dict(pd.json_normalize(information))
+    df.to_csv("notebooks/"+ csv_file_name+".csv")
```

Comment:
the prefix `"notebooks/"`shouldn´t be hardcoded inside the function, because otherwise it´s not reusable